festim/species.py

```diff
@@ -10,6 +13,7 @@ class Species:
         solution (dolfinx.fem.Function or ...): the solution for the current timestep
         prev_solution (dolfinx.fem.Function or ...): the solution for the previous timestep
         test_function (ufl.Argument or ...): the testfunction associated with this species
+        concentration (dolfinx.fem.Function): the concentration of the species
```

Comment:
does the species need a concentration attribute too?

Reply:
This can be a bit redundant since for `Species` the concentration is `solution`. Although it provides the users a more straightforward way to get the concentration.

But it mostly to slightly simplify the implementation of `Reaction.reaction_term()`:

https://github.com/RemDelaporteMathurin/FESTIM/blob/ec2d15ea9bd7764a0aef22a979ed9b54cf39f89c/festim/reaction.py#L80-L81